src/ledger/meta.zig

```diff
@@ -258,12 +260,18 @@ pub const ErasureConfig = struct {
 /// Index recording presence/absence of shreds
 pub const Index = struct {
     slot: Slot,
+    // QUESTION: maybe name these data_index and code_index?
+    // while reading the code, I often thought this was refering to the actual data shred and code shred
+    // but it is just the index.
     data: ShredIndex,
     code: ShredIndex,
 
     pub fn init(allocator: std.mem.Allocator, slot: Slot) Index {
         return .{
             .slot = slot,
+            // QUESTION: So the slot is passed in here, but it is not used to construct
+            // the data and code index? Would have expected some relationship here between
+            // the slot being passed and the data/code index?
```

Comment:
The indexes tell you which shreds have been received for a particular slot. The slot number is an input here just to include a field that specifies which slot this index is for. But the slot number itself doesn't tell you which shreds have been received for that slot.

Maybe this struct doesn't even need the slot field at all. Current usages would require some rework to get the data from somewhere else though.